exchange/build_test.wasm.go

```diff
@@ -0,0 +1,310 @@
+//go:build wasm
```

Comment:
A bit confused. So these test files will only be run on `wasm` builds?

Why test files require a build tag?

Reply:
the test files do not require build tags  removed those :+1: